### DISCUSSION TITLE
What is the use case of `vanX.stateFields`?

### DISCUSSION THREAD

**Discussion by jeffreyyoung** [vanjs-org/van#236] [Q&A] [answered]
I noticed this function in the docs, but from reading the docs, I'm not sure on when it's useful.

I'm really enjoying vanJS, thanks for the library!
https://vanjs.org/x#api-stateFields

**Comment by Tao-VanJS** [comment, marked as answer]
The underlying mechanism of reactive object is to store a collection of `State` objects (which can be deeply nested as well). `vanX.stateFields` is helpful to retrieve these underlying `State` objects so that you can pass them to places where `State` objects are expected.
